include/apq/connection_info.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <apq/connection.h>
+#include <apq/impl/async_connect.h>
+
+#include <chrono>
+
+namespace libapq {
+namespace detail {
+
+template <typename Handler, typename Connection>
+struct connection_handler_binder {
+    Handler handler_;
+    Connection conn_;
+
+    void operator() (error_code ec) {
+        handler_(std::move(ec), std::move(conn_));
+    }
+
+    template <typename Func>
+    friend void asio_handler_invoke(Func&& f, connection_handler_binder* ctx) {
+        using ::boost::asio::asio_handler_invoke;
+        asio_handler_invoke(std::forward<Func>(f), 
+            std::addressof(ctx->handler_));
+    }
+};
+
+template <typename Handler, typename Connection>
+inline auto bind_connection_handler(Handler&& base, Connection&& conn) {
+    return connection_handler_binder<std::decay_t<Handler>, std::decay_t<Connection>> {
+        std::forward<Handler>(base), std::forward<Connection>(conn)
+    };
+}
+
+} // namespace detail
+
+template <
+    typename OidMap = empty_oid_map, 
+    typename Statistics = no_statistics>
+class connection_info {
+    io_context& io_;
+    std::string raw_;
```

Comment:
Does connection_info require to own connection info string? Why don't just use string_view?

Reply:
In case of string_view usage, our user will be forced to store this string somewhere else and have a lot of fun with string lifetime. This object is a connection producer, so it is not expected to create/copy it often. I do not see any good reason to perform such optimisation here. Do you?

Reply:
I don't, ok.